src/fonts.js

```diff
@@ -130,7 +130,10 @@ export const fontsDialogPlugin = (editor, opts) => {
     // update the head and the ui when the frame is loaded
     editor.on('canvas:frame:load', () => refresh(editor, opts))
     // When the page changes, update the dom
-    editor.on('page', () => refresh(editor, opts))
+    editor.on('page', () => {
+        // FIXME: remove this timeout which is a workaround for issues with fonts loading after page change
+        setTimeout(() => refresh(editor, opts), 50)
```

Comment:
This is great if it works, but would you try this other solution? When the page event is fired, use editor.once to listen to the event `canvas:frame:load` instead of setTimeout

Or maybe you tried already and it didn't work?

Reply:
I will test it out on Monday and let you know
there is no hurry, lets take our time to make sure we fix it properly :)

Reply:
Just checked. Unfortunately this does not fix the issue:

```
editor.on('page', () => {
    editor.once('canvas:frame:load', () => {
        refresh(editor, opts)
    }
})
```

I added a `console.log` and the once event is fired, but aparently too early to fix the font loading issue.

Reply:
Also - this was happening on my sandbox website, which was just for testing various features of Silex, and it's a bit of a mess (I did a lot of page cloning, there are probably a lot of orphaned styles).

I have started a new website where I carefully design the actual layout I want, and on that project the problem has not yet happened. I wonder why though :thinking: I'll update here if I learn anything new.